File: app/src/main/java/com/tpeapp/handler/ChatMessage.kt
package com.tpeapp.handler

/**
 * A single message in the conversation with "Handler", the AI companion.
 *
 * @param role      "user" or "assistant" — matches the OpenAI Chat Completions role field.
 * @param content   The message body.
 * @param timestamp Unix epoch ms when the message was created.
 */
data class ChatMessage(
    val id: String,
    val role: String,
    val content: String,
    val timestamp: Long
) {
    val isUser: Boolean get() = role == "user"
}

File: flutter_app/lib/models/chat_message.dart
/// Mirrors [com.tpeapp.handler.ChatMessage].
class ChatMessage {
  const ChatMessage({
    required this.id,
    required this.role,
    required this.content,
    required this.timestamp,
  });

  final String id;

  /// `"user"` or `"assistant"`.
  final String role;
  final String content;

  /// Unix epoch milliseconds.
  final int timestamp;

  bool get isUser => role == 'user';

  factory ChatMessage.fromJson(Map<String, dynamic> json) => ChatMessage(
        id: json['id'] as String,
        role: json['role'] as String,
        content: json['content'] as String,
        timestamp: json['timestamp'] as int,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'role': role,
        'content': content,
        'timestamp': timestamp,
      };
}
